File: C++/0119.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        int prev = 1;
        vector<int> ans = {1};

        for(int i = 1; i <= rowIndex; i++){
            long cur = prev * (rowIndex - i + 1) / i;
            ans.push_back(cur);
            prev = cur;
        }
        return ans;
    }
};

// Time Complexity: O(n)
// Space Complexity: O(n)

File: Python/0119.py
class Solution:
    def getRow(self, rowIndex: int) -> list[int]:
        prev = 1
        ans = [1]
        
        for i in range(1, rowIndex+1):
            cur = prev * (rowIndex-i+1) // i
            ans.append(cur)
            prev = cur

        return ans

# Time Complexity: O(n)
# Space Complexity: O(n)
